feat: context propagation (#227)
open-feature/spec: specification/glossary.md

Context object for flag evaluation, which may contain information about the runtime environment, details of the transport method encapsulating the flag evaluation, the host, the client, the subject (user), etc. This data may be used as a basis for differential evaluation of feature flags based on rules that can be defined in the flag system. Context data may be provided by merging static global context, arguments to flag evaluation, and implicit language-dependant state propagation mechanisms (thread-local storage, promise chains, continuations, etc).

### Transaction Context Propagator

> An SDK-compliant implementation which stores and returns transaction specific evaluation context. A transaction can be something like a thread or web request.

Reviewer: ```suggestion
An SDK-compliant implementation which stores and returns transaction-specific evaluation context. A _transaction_ might be a web request or application event, which carries it's own contextual data in thread or continuation storage.
```

Minor change here, but I tried to add two examples of a "transaction" and separately, two examples of a storage mechanism. They are slightly different ideas.

> An SDK-compliant implementation which stores and returns transaction-specific evaluation context. A _transaction_ might be a web request or application event, which carries it's own contextual data in thread or continuation storage.

Reviewer 2: ```suggestion
An SDK-compliant implementation that stores and returns transaction-specific evaluation context. A _transaction_ might be a web request or application event, which carries its contextual data in a thread or continuation storage.
```
